src/Service.php

```diff
@@ -7,135 +7,287 @@
-		$methods = [];
-		foreach ($this->getMethods() as $_method) {
-			if ($this->isMethodCovered($_method)) {
-				$methods[$_method] = $_method;
-			}
+		//pokud je pole prazdne, je sance ze se jeste nic nehledalo -> prohledame
+		if (empty($this->foundTests)) {
+			$this->findAvailableTests();
+			$this->checkCoverage();
 		}
-
-		return $methods;
+		return $this->foundTests;
 	}
 
+
+	/**
+	 * Vraci pole obsahujici testovaci metody, ktere se nepodarilo nalezt a jsou treba k pokryti
+	 * @throws \ReflectionException
+	 */
 	public function getMissingMethods(): array
 	{
-		$methods = [];
-		foreach ($this->getMethods() as $_method) {
-			if (! $this->isMethodCovered($_method)) {
-				$methods[$_method] = $_method;
-			}
+		//prazdne pole -> radeji prohledame
+		if (empty($this->missingTests)) {
+			$this->findAvailableTests();
+			$this->checkCoverage();
 		}
+		return $this->missingTests;
+	}
+
+
+	/**
+	 * Kontrola zda se jedna o metodu, ktera je urcena nastavenim k otestovani.
+	 * @param string $methodName
+	 * @param string|null $prefix
+	 */
+	protected static function isMethodToTest(string $methodName, string $prefix = null) : bool
+	{
+		if (Strings::startsWith($methodName, $prefix)) {
+			return true;
+		}
+		return false;
+	}
+
+
+	/**
+	 * metoda kontroluje pokryti metod testy.
+	 * @throws \ReflectionException
+	 */
+	public function checkCoverage() {
+		$cwd = getcwd().'/';
+		if (!empty($foundTests) || !empty($missingTests)) {
+			return;
+		}
+
+		$this->findMethodsToCover();
+		foreach ($this->methodsToCover as $key => $method) {
 
-		return $methods;
+			//ziskame cast namespace podle ktere porovnavame
+			$position = strpos($method, '\\');
+			if ($position) {
+				$method = substr($method, $position + 1);
+			}
+
+			if (array_key_exists($method, $this->existingTests)) {
+				$this->foundTests[] = str_replace($cwd, '', $this->existingTests[$method]);
+			} else {
+				$this->missingTests[] = str_replace($cwd, '', $key);
+			}
+		}
 	}
 
-	public function getUrls(?string $prefix = null): array
+
+	/**
+	 * Nacteni konfigurace pro vyheldavani a nacteni vsech souboru ktere jsou zadany
+	 */
+	public function getRobotLoader(): RobotLoader
 	{
-		$urls = [];
-		foreach ($this->getFoundMethods() as $method) {
-			if ($prefix && !Strings::startsWith($method, $prefix)) {
-				continue;
+		if (!$this->robotLoader) {
+			$this->robotLoader = (new RobotLoader)
+				->setTempDirectory($this->config['tempDir']);
+
+			// iterace pres vsechny nakonfigurovane slozky
+			foreach ($this->config['componentCoverage'] as $key => $dirSetup) {
+
+				// kontrola ze mame nakonfigurovany vsechny potrebne udaje
+				// budeme uzivatele informovat o tom ze neco nema dobre nakonfigurovane
+				if(!array_key_exists('componentDir', $dirSetup)
+					|| !array_key_exists('fileMask', $dirSetup)
+					|| !array_key_exists('methodMask', $dirSetup)
+					|| !array_key_exists('testDir', $this->config)) {
+
+					$this->skippedForMissingConfiguration[] = $key;
```

Comment:
Nechceme tady místo `key` dát nějakou hlášku co přesně tam chybí, ať je to pro programátora jasnější co je blbě.